tests/test_beam.py

```
@@ -952,6 +952,52 @@ def test_sleep_consolidated_content_is_recallable(self, temp_db, monkeypatch):
+        second = Mnemosyne(session_id="second", db_path=temp_db)
+
+        first.close()
+        assert second.beam.conn.execute("SELECT 1").fetchone()[0] == 1
+        second.close()
+
+    def test_constructor_failure_releases_provisional_connection(self, temp_db, monkeypatch):
+        import mnemosyne.core.memory as memory_module
+
+        real_beam = memory_module.BeamMemory
+
+        def fail_beam(*args, **kwargs):
```

Ruff (0.15.21)

[warning] Missing return type annotation for private function `fail_beam`

Add return type annotation: `NoReturn`

(ANN202)